File: rev6.py
my_dict = {
        "greg": "mathias",
        "mike": "malone",
        "jim": "malone",
        "jim": "smith"
        }

def main():
    reverse = make_rev(my_dict)
    print(my_dict)
    print(reverse)

def make_rev(data):
    rev_data = {}
    for old_key in data:
        old_value = data[old_key]
        if old_value not in rev_data:
            rev_data[old_value] = [old_key]
        else:
            rev_data[old_value].append(old_key)
    return rev_data

main()

File: rev6.js
const reverse = () => {
	let revDict = {}
	const myDict = {
		"banana": "yellow",
		"peach": "yellow",
		"apple": "red",
		"strawberry": "red"
	}
	for (const old_key in myDict) {
		let old_value = myDict[old_key]
		if (old_value in revDict) {
			revDict[old_value].push(old_key)
		} else {
			revDict[old_value] = [old_key]
		}

	}
	return console.log(revDict)
}

reverse()
